contrib/qpl-cmake/CMakeLists.txt

```diff
@@ -0,0 +1,321 @@
+## The Intel® QPL provides high performance implementations of data processing functions for existing hardware accelerator, and/or software path in case if hardware accelerator is not available.
+if (OS_LINUX AND ARCH_AMD64 AND (ENABLE_AVX2 OR ENABLE_AVX512))
```

Comment:
Just for my understanding: [0] says that the SW path needs either AVX2 or AVX512 and that the HW path needs the "Intel® In-Memory Analytics Accelerator" (IAA) device. Could you share if this device is bundled with recent Intel processors (and if yes, which) or if it's a separate offloading device connected e.g. via PCIe, CXL etc.? And how widespread is it already in the market? I did not find too much information online.

Also: Could you please confirm that QPL detects presence of IAA and uses it automatically? (i.e. there is nothing which needs to be configured in ClickHouse's build or runtime for IAA to work).

[1] mentions that `Decompression is supported for Deflate streams where the size of the history buffer is no more than 4 KB. (The default size for Deflate is 32 KB.) Using an input stream with a larger history size results in an error.`. Is it right that this only applies to the HW path and that data DEFLATE-compressed by the SW-path (or another non-QPL SW implementation) can potentially not be uncompressed by the HW path? If so, this should at least be mentioned in the docs for DEFLATE.

[0] https://intel.github.io/qpl/documentation/introduction_docs/system_requirements.html
[1] https://d2pgu9s4sfmw1s.cloudfront.net/UAM/Prod/Done/a062E00001UMAZeQAP/414fecc9-85ff-43ee-b83d-033be5be5ac5?Expires=1657094154&Key-Pair-Id=APKAJKRNIMMSNYXST6UA&Signature=K13uuyxar7osZev5Gxtipv6xdM-4E8fsLFXFloQEb0feMVsVJLug6eI4UNTwua0SOKHgcJYD-tb4KWXUsxiBfYNBn6kNbcOtbUVTnKaYdsbFdhWQhrWn9rNkG7XiMb2QNsUxta8K4RzdhvtTR3YXLDksG73AnSA1EPkydcMDSRifPbDcKQPf-~Wmv4cJGSO7ThAh5ZzXmM2~B5MLWW-b6wAt4-eM8HdMhZKHLRXPWQAXUYGcC2YoXsZDT7Ab8WKkQCmhN2NVDQRJ62sZHotfHH4pr-QMMiUpi0yLacd2loAiUczQhCmBiW-Qw6~mZxtbzGTVQ5tMRnUp8fUcnl0sDg__

Reply:
@rschu1ze  
for the question: Could you share if this device is bundled with recent Intel processors (and if yes, which) or if it's a separate offloading device connected e.g. via PCIe, CXL etc.? And how widespread is it already in the market? 
The Intel® In-Memory Analytics Accelerator (Intel® IAA) is a hardware accelerator that provides very high
throughput compression and decompression combined with primitive analytic functions supported in Next Generation Intel® Xeon® Scalable processors, codenamed Sapphire Rapids processor.  it's an accelerator integrated in CPU and it's not separate device.  we have a spec here: https://cdrdv2.intel.com/v1/dl/getContent/721858

Reply:
Thanks, that was insightful (also the spec).

About not being able to copy data between nodes with and without IAA (i.e. SW path vs. HW path) - I think that is overall okay ***if*** we put a word of caution into ClickHouse's documentation. The right place would be here: https://clickhouse.com/docs/en/sql-reference/statements/create/table/#general-purpose-codecs

Reply:
@rschu1ze 
Need clarify further:
Data DEFLATE-compressed by the QPL SW-path can be uncompressed by the QPL HW-path, and vice versa.
DEFLATE-compressed by non-QPL SW implementation, such as ZLIB, can potentially not be uncompressed by the both of QPL HW and SW path because QPL limit history buffer to be 4KB while the default size for Deflate is 32 KB.

Reply:
> Thanks, that was insightful (also the spec).
> 
> About not being able to copy data between nodes with and without IAA (i.e. SW path vs. HW path) - I think that is overall okay _**if**_ we put a word of caution into ClickHouse's documentation. The right place would be here: https://clickhouse.com/docs/en/sql-reference/statements/create/table/#general-purpose-codecs

Done